File: src/main/java/com/aries/tutorial/batch/InstanceMetricsBatch.java
package com.aries.tutorial.batch;

import com.aries.extension.data.BatchData;
import com.aries.extension.data.batch.MetricsDataAsInstance;
import com.aries.extension.handler.BatchHandler;
import com.aries.extension.util.PropertyUtil;
import com.aries.extension.util.LogUtil;

public class InstanceMetricsBatch implements BatchHandler {
    @Override
    public boolean preHandle(long batchTime) {
        // TODO: Adding pre-processing code
        return true;
    }

    @Override
    public void process(BatchData[] batchData) {
        LogUtil.info("[InstanceMetricsBatch] - " +
                PropertyUtil.getValue("instance_metrics_batch", "subject", "Unknown subject"));

        for(int i = 0; i < batchData.length; i++) {
            MetricsDataAsInstance data = (MetricsDataAsInstance) batchData[i];

            StringBuilder sb = new StringBuilder();
            sb.append("Domain ID : " + data.domainId + ", ");
            sb.append("Domain Name : " + data.domainName + ", ");
            sb.append("Service Count : " + data.serviceCount + ", ");
            sb.append("Max TPS : " + data.maxTps + ", ");
            sb.append("Active Service : " + data.maxTps + ", ");
            sb.append("Error Count : " + data.errorCount);

            LogUtil.info(sb.toString());
        }
    }
}

File: src/main/kotlin/com/aries/tutorial2/batch/InstanceMetricsBatch.kt
package com.aries.tutorial2.batch

import com.aries.extension.data.BatchData
import com.aries.extension.data.batch.MetricsDataAsInstance
import com.aries.extension.handler.BatchHandler
import com.aries.extension.util.LogUtil
import com.aries.extension.util.PropertyUtil

class InstanceMetricsBatch : BatchHandler {
    override fun preHandle(batchTime: Long): Boolean {
        // TODO: Adding pre-processing code
        return true
    }

    override fun process(batchData: Array<BatchData>) {
        LogUtil.info("[InstanceMetricsBatch] - " + PropertyUtil.getValue("instance_metrics_batch", "subject", "Unknown subject"))

        for (d in batchData) {
            val data = d as MetricsDataAsInstance

            LogUtil.info("Domain ID : " + data.domainId)
            LogUtil.info("Instance Name : " + data.instanceName)
            LogUtil.info("Call Count : " + data.serviceCount)
            LogUtil.info("Max TPS : " + data.maxTps)
            LogUtil.info("Active Service : " + data.activeService)
            LogUtil.info("Error Count : " + data.errorCount + "\n")
        }
    }
}
